packages/block-library/src/image/edit.js

```diff
@@ -364,35 +380,72 @@ export function ImageEdit( {
 	};
 
 	return (
-		<figure { ...blockProps }>
-			<Image
-				temporaryURL={ temporaryURL }
-				attributes={ attributes }
-				setAttributes={ setAttributes }
-				isSingleSelected={ isSingleSelected }
-				insertBlocksAfter={ insertBlocksAfter }
-				onReplace={ onReplace }
-				onSelectImage={ onSelectImage }
-				onSelectURL={ onSelectURL }
-				onUploadError={ onUploadError }
-				context={ context }
-				clientId={ clientId }
-				blockEditingMode={ blockEditingMode }
-				parentLayoutType={ parentLayout?.type }
-			/>
-			<MediaPlaceholder
-				icon={ <BlockIcon icon={ icon } /> }
-				onSelect={ onSelectImage }
-				onSelectURL={ onSelectURL }
-				onError={ onUploadError }
-				placeholder={ placeholder }
-				accept="image/*"
-				allowedTypes={ ALLOWED_MEDIA_TYPES }
-				value={ { id, src } }
-				mediaPreview={ mediaPreview }
-				disableMediaButtons={ temporaryURL || url }
-			/>
-		</figure>
+		<>
+			<figure { ...blockProps }>
+				{ resizeListener }
+				{ ! ( temporaryURL || url ) && (
+					<BlockControls group="other">
+						<MediaReplaceFlow
```

Comment:
Is it possible to integrate this code into [the existing `MediaReplaceFlow` component](https://github.com/WordPress/gutenberg/blob/00d1f8afac2e3b9764dc28261462b72794ae5ce0/packages/block-library/src/image/image.js#L590-L602)?

The reason is that, for example, if multiple blocks are selected or if the URL control is locked, the `MediaReplaceFlow` component should not be rendered.

For example, try inserting the following HTML via the code editor:

```html
<!-- wp:image { "metadata":{ "bindings":{ "url":{ "source":"core/post-meta", "args":{ "key":"test" } } } } } -->
<figure class="wp-block-image"><img alt=""/></figure>
<!-- /wp:image -->
```


In this case, the "Add image" control should not be displayed:

![image](https://github.com/user-attachments/assets/d19c0445-9a7d-4bdf-9018-a5b929960cf9)

The existing `MediaReplaceFlow` component has checks for these:

https://github.com/WordPress/gutenberg/blob/00d1f8afac2e3b9764dc28261462b72794ae5ce0/packages/block-library/src/image/image.js#L590

Furthermore, in this PR, it seems that it is no longer possible to replace images by dropping them, as shown by the E2E test failure. This might be related to the fact that the `MediaReplaceFlow` component is separated:

https://github.com/user-attachments/assets/0df35c35-981e-4825-9c62-230d456d7e0c